pkg/keys/ecdsa.go

```diff
@@ -4,64 +4,137 @@ import (
 	"bytes"
 	"crypto/ecdsa"
 	"crypto/elliptic"
+	"crypto/rand"
 	"crypto/sha256"
-	"encoding/asn1"
+	"crypto/x509"
 	"encoding/json"
+	"encoding/pem"
 	"errors"
 	"fmt"
 	"io"
-	"math/big"
+	"os"
 
 	"github.com/theupdateframework/go-tuf/data"
 )
 
 func init() {
 	VerifierMap.Store(data.KeyTypeECDSA_SHA2_P256, NewEcdsaVerifier)
+	SignerMap.Store(data.KeyTypeECDSA_SHA2_P256, newEcdsaSigner)
 }
 
 func NewEcdsaVerifier() Verifier {
-	return &p256Verifier{}
+	return &ecdsaVerifierWithDeprecatedSupport{}
 }
 
-type ecdsaSignature struct {
-	R, S *big.Int
+func newEcdsaSigner() Signer {
+	return &ecdsaSigner{}
 }
 
-type p256Verifier struct {
+type ecdsaVerifierWithDeprecatedSupport struct {
+	key *data.PublicKey
+	// This will switch based on whether this is a PEM-encoded key
+	// or a deprecated hex-encoded key.
+	Verifier
+}
+
+func (p *ecdsaVerifierWithDeprecatedSupport) UnmarshalPublicKey(key *data.PublicKey) error {
+	p.key = key
+	pemVerifier := &ecdsaVerifier{}
+	if err := pemVerifier.UnmarshalPublicKey(key); err != nil {
```

Comment:
Guarding behind a flag would be great! I could imagine a build tag too that only adds the deprecated if the tag is enabled. But this API can be called as a library, and not as a compiled package. So maybe some options are:
* An environment variable, go-tuf specific: `GO_TUF_ENABLE_DEPRECATED_ECDSA`
* Adding an `init` function for deprecated keys into a separate package, and not including that into the default binary. Then only libraries would be able to use it on import.

Reply:
I like the second one, if it's not too much work.

Reply:
Me too! All set. I have to add a separate package `pkg/deprecated` so that the init function only runs when someone pull this in (as opposed to the keys package). Added a repo level test in there so that you can verify repository metadata with hex-encoded keys. Modified the `repo_test.go` so that it *fails* with the hex-encoded keys (same test, just failure)